src/event/views/CreateEventPage.vue

```diff
@@ -131,24 +154,58 @@ export default defineComponent({
     DatePicker,
     DayPicker,
     TimePicker,
-    TextInput
+    TextInput,
+    LoginPage
   },
-  data() {
-    return {
-      isHidden: true,
-      isDatePickerEvent: true,
-      startTime: set(new Date(), { hours: 9, minutes: 0 }),
-      endTime: set(new Date(), { hours: 21, minutes: 0 }),
-      timezone: "America/Vancouver",
-      eventTitle: "",
-      eventDescription: "",
-      isChecked: true
-    };
+  data: function() {
+    return initialState();
   },
-  computed: {},
   methods: {
     toggleEventType(toggleState: boolean) {
       this.isDatePickerEvent = toggleState;
+    },
+    showModal() {
+      this.isModalVisible = true;
+    },
+    closeModal() {
+      this.isModalVisible = false;
+    },
+    reset() {
+      Object.assign(this.$data, initialState());
+    },
+    async createEvent(e: any) {
+      e.preventDefault();
+      if (!Auth.currentUser) {
+        console.log("not logged in!");
+        this.showModal();
+        return;
+      } else {
+        console.log("logged in!");
+        const event: Event = {
+          users: [],
+          owners: [Auth.currentUser.uid],
+          scheduleWindow: {
+            startTime: this.startTime,
+            endTime: this.endTime
+          },
+          calendar: {
+            blocks: []
+          },
+          title: this.eventTitle,
+          timezone: this.timezone,
+          description: this.eventDescription
+        };
+        client
+          .addEvent(event)
+          .then(eventId => {
+            console.log("created new event with id: " + eventId);
+            this.$router.push(`/event/${eventId}`);
+          })
+          .catch(err => {
+            console.log("could not create event: " + err);
+          });
+        this.reset();
```

Comment:
I think this should be moved inside of the `.then` block since the `addEvent` call wouldn't have finished by this point. Or maybe remove it all together since we're redirecting the user nevertheless.

Reply:
I was thinking of adding a simple error popup/modal component (perhaps under the common/ folder) that we could open up if there's any error with `client.addEvent` for whatever reason. so maybe something where we could pass in an error message, that may be reusable among different components that make API calls? would that be useful/helpful, or no need (cause it doesn't feel like among the most important features we could implement given the time left)?